Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -134,20 +134,50 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
-            top_right.horizontal_radius().to_string(),
-            bottom_right.horizontal_radius().to_string(),
-            bottom_left.horizontal_radius().to_string(),
-            top_left.vertical_radius().to_string(),
-            top_right.vertical_radius().to_string(),
-            bottom_right.vertical_radius().to_string(),
-            bottom_left.vertical_radius().to_string()));
+        auto top_left = longhand(PropertyID::BorderTopLeftRadius);
+        auto top_right = longhand(PropertyID::BorderTopRightRadius);
+        auto bottom_right = longhand(PropertyID::BorderBottomRightRadius);
+        auto bottom_left = longhand(PropertyID::BorderBottomLeftRadius);
+
+        auto horizontal_radius = [&](auto& style_value) -> String {
+            if (style_value->is_border_radius())
+                return style_value->as_border_radius().horizontal_radius().to_string();
+            return style_value->to_string(mode);
+        };
+
+        auto top_left_horizontal_string = horizontal_radius(top_left);
+        auto top_right_horizontal_string = horizontal_radius(top_right);
+        auto bottom_right_horizontal_string = horizontal_radius(bottom_right);
+        auto bottom_left_horizontal_string = horizontal_radius(bottom_left);
+
+        auto vertical_radius = [&](auto& style_value) -> String {
+            if (style_value->is_border_radius())
+                return style_value->as_border_radius().vertical_radius().to_string();
+            return style_value->to_string(mode);
+        };
+
+        auto top_left_vertical_string = vertical_radius(top_left);
+        auto top_right_vertical_string = vertical_radius(top_right);
+        auto bottom_right_vertical_string = vertical_radius(bottom_right);
+        auto bottom_left_vertical_string = vertical_radius(bottom_left);
+
+        auto serialize_radius = [](auto& top_left, auto& top_right, auto& bottom_right, auto& bottom_left) -> String {
+            if (first_is_equal_to_all_of(top_left, top_right, bottom_right, bottom_left))
+                return top_left;
+            if (top_left == bottom_right && top_right == bottom_left)
+                return MUST(String::formatted("{} {}", top_left, top_right));
+            if (top_right == bottom_left)
+                return MUST(String::formatted("{} {} {}", top_left, top_right, bottom_right));
+
+            return MUST(String::formatted("{} {} {} {}", top_left, top_right, bottom_right, bottom_left));
+        };
+
+        auto first_radius_serialization = serialize_radius(top_left_horizontal_string, top_right_horizontal_string, bottom_right_horizontal_string, bottom_left_horizontal_string);
+        auto second_radius_serialization = serialize_radius(top_left_vertical_string, top_right_vertical_string, bottom_right_vertical_string, bottom_left_vertical_string);
+        if (first_radius_serialization == second_radius_serialization)
+            return MUST(String::formatted("{}", first_radius_serialization));
```

Comment:
```suggestion
            return first_radius_serialization;
```

Reply:
I get a warning when I do this that: "The address of the local temporary object may escape the function". I've changed `serialize_radius` to take `StringView` instead and I create the new string there instead of here. There's probably a better way to do this that avoid an unnecessary string creation :shrug:

Reply:
That's odd, wasn't the code entirely dealing with Strings before? I don't see where it'd be losing anything. Though I'm not the best at lifetimes.

Reply:
I'm also not fantastic at this stuff, but I've now done it in a way that should avoid any unnecessary String copies and avoid the warning.